src/test/optimizer/strategy/between_composition_rule_test.cpp

```diff
@@ -0,0 +1,258 @@
+
+  const auto expected_lqp = input_lqp;
+  // clang-format on
+
+  const auto result_lqp = StrategyBaseTest::apply_rule(_rule, input_lqp);
+
+  EXPECT_LQP_EQ(result_lqp, expected_lqp);
+}
+
+TEST_F(BetweenCompositionTest, ImpossibleColumnRange) {
+  // clang-format off
+  const auto input_lqp = PredicateNode::make(
+      greater_than_equals_(a, 300),
+      PredicateNode::make(
+        less_than_equals_(a, 200),
+          node));
+
+  const auto expected_lqp = PredicateNode::make(between_(a, 300, 200), node);
+  // clang-format on
+
+  const auto result_lqp = StrategyBaseTest::apply_rule(_rule, input_lqp);
+
+  EXPECT_LQP_EQ(result_lqp, expected_lqp);
+}
+
+TEST_F(BetweenCompositionTest, LeftExclusive) {
+  // clang-format off
+  const auto input_lqp = PredicateNode::make(
+      greater_than_(a, 200),
+      PredicateNode::make(
+        less_than_equals_(a, 300),
+          node));
+
+  const auto expected_lqp = PredicateNode::make(
+      std::make_shared<BetweenExpression>(
+        a.original_node()->column_expressions()[0],
+        value_(200),
+        value_(300),
+        false,
+        true),
+      node);
+  // clang-format on
+
+  const auto result_lqp = StrategyBaseTest::apply_rule(_rule, input_lqp);
+
+  EXPECT_LQP_EQ(result_lqp, expected_lqp);
+}
+
+TEST_F(BetweenCompositionTest, RightExclusive) {
+  // clang-format off
+  const auto input_lqp = PredicateNode::make(
+      greater_than_equals_(a, 200),
+      PredicateNode::make(
+        less_than_(a, 300),
+          node));
+
+  const auto expected_lqp = PredicateNode::make(
+      std::make_shared<BetweenExpression>(
+        a.original_node()->column_expressions()[0],
+        value_(200),
+        value_(300),
+        true,
+        false),
+      node);
+  // clang-format on
+
+  const auto result_lqp = StrategyBaseTest::apply_rule(_rule, input_lqp);
+
+  EXPECT_LQP_EQ(result_lqp, expected_lqp);
+}
+
+TEST_F(BetweenCompositionTest, BothExclusive) {
+  // clang-format off
+  const auto input_lqp = PredicateNode::make(
+      greater_than_(a, 200),
+      PredicateNode::make(
+        less_than_(a, 300),
+          node));
+
+  const auto expected_lqp = PredicateNode::make(
+      std::make_shared<BetweenExpression>(
+        a.original_node()->column_expressions()[0],
+        value_(200),
+        value_(300),
+        false,
+        false),
+      node);
+  // clang-format on
+
+  const auto result_lqp = StrategyBaseTest::apply_rule(_rule, input_lqp);
+
+  EXPECT_LQP_EQ(result_lqp, expected_lqp);
+}
+
+TEST_F(BetweenCompositionTest, FindOptimalBetween) {
+  // clang-format off
+  const auto input_lqp = PredicateNode::make(
+      greater_than_equals_(a, 100),
+      PredicateNode::make(
+        greater_than_equals_(a, 200),
+        PredicateNode::make(
+          less_than_equals_(a, 400),
+          PredicateNode::make(
+            less_than_equals_(a, 300),
+            node))));
+
+  const auto expected_lqp = PredicateNode::make(between_(a, 200, 300), node);
```

Comment:
Until now, I had no idea that your rule did this. *Please* improve its documentation.